### DISCUSSION TITLE
Resource pinning not auto triggering a new build

### DISCUSSION THREAD

**Discussion by fauna-chase** [concourse/concourse#6402] [Help & Support]
Are there scenarios where pinning a resource that is set to trigger a job to a version that is not the previous build is not expected to trigger the job?

I'm running into a situation where I pin a resource that is a trigger, it is different than the previous build, and it does not automatically trigger the job to run.  The job has been previously run with the pinned version, but it wasn't the most recent version that the job was run with.  Based on these docs, https://concourse-ci.org/scheduler.html#scheduling-behavior, I'd expect it to, has me wondering if there is something the algorithm is doing to not select it as a candidate version.  Would appreciate some help to better understand the scenario.

Thanks!
Chase

**Comment by taylorsilva** [comment]
This looks like a bug based on what the docs describe. I confirmed the behaviour you're seeing with this pipeline:

```yaml
resources:
- name: time-resource
  type: time
  source:
    interval: 10s

jobs:
- name: some-job
  plan:
  - get: time-resource
    trigger: true
```

I let it create a few versions (four) and go through one build for each version. Then I pinned the version for the 2nd build. Two things happened:
- the job never triggered
- when the next resource check ran the order of the versions changed. The version I had pinned, the 2nd version for build 2, jumped to the second position in the list of versions. Not sure if this is a quirk of the time resource or something we're doing in the backend/algorithm. Either way, it was unexpected to me.

image showing the versions are out of order:
<img width="367" alt="image" src="https://user-images.githubusercontent.com/4583837/104039627-9dde2580-51a4-11eb-9a6a-3346602b6693.png">


@vito @clarafu is this a bug or are the docs not aligned with new algorithm behaviour? I'm guessing the former but want confirmation from y'all please since you're the experts in that area 😅

**Reply by fauna-chase** [reply]
Appreciate you digging into this! I will say that for non time based resources I don't see the version order changing

**Reply by taylorsilva** [reply]
User on discord pointed me to this commit: https://github.com/concourse/concourse/commit/6262a3c0d3df0b7edc81bccb0d28e7f89fc2677f

HMMM, looks like our docs are outdated. The shifting versions thing may still be an issue, but not 100% sure. Clara and Vito aren't around to answer anymore, and like me, they probably don't remember. Going to just document what the code is doing as it currently stands.
